### ISSUE TITLE
Stop/Undo sharing code between new & old open

### ISSUE THREAD

**Issue by ChristopherChudzicki** [mitodl/open-discussions#3852]
tldr: Initially we wanted to "share" code between the old/new frontend sections. Subsequently we decided to **move** features from old to new. Some code is stilled shared, and this makes changing things harder. Let's stop sharing.

---

The frontend portion of the MIT Open codebase is split between "new" Typescript and "old" flow files. The new TS files are organized as several local node packages.

When we began working on new features for Open, our initial plan was to migrate code as needed from the old flow js files to separate local packages and new TS version in both "old" and "new" open. An example of this is the Searchbar component: It is written in Typescript in `ol-search-ui` package; the compiled JS is used to provide the searchbar both at https://discussions-rc.odl.mit.edu/learn/search and https://discussions-rc.odl.mit.edu/infinite/search. 

Sharing code in this way between new features and old features proved difficult[^1]. We decided to stop sharing: new TS code should only be used in "new" open, and we will gradually move code from the JS to TS codebases. 

[^1]: Example: old codebase uses a fair amount of redux, largely for storing api requests, but also for some other things. New codebase does not. Writing a shared component that worked well in both codebases would be hard. More generally, a stricter separation allows us to make separate decisions about library-usage in the new codebase.

We should remove the old shared code. Two benefits:
1. We can iterate on new open without worrying about breaking old open features
2. We will no longer need to compile individual local packages from TS to JS, simplifying development.

**Comment by pdpinch** [comment]
What's the scope of this? Can it be broken down into smaller chunks?

**Comment by ChristopherChudzicki** [comment]
@pdpinch The shared components are:
1. The searchbox from `ol-search-ui` package
2. ~7 simple utility layout components from `ol-util` (e.g, Card, the "hero area" banner of pages)
3. A simple component for validation errors from `ol-forms`

I would expect the changes to maybe 2 days overall. I think doing each of 1, 2, 3 as a separate PR makes sense for easier reviews.

A followup chunk of work would be "Stop compiling individual packages from TS to JS".

**Comment by pdpinch** [comment]
Can you say more about the follow-up chunk? How big of a lift is that, and what are the benefits?

**Comment by ChristopherChudzicki** [comment]
The followup chunk would be simple changes: Changing the `package.json` exports to point to typescript files instead of JS files. The benefit I see is simpler build: Right now, we need to be sure all the local typescript packages are built before the "new open" webpack build runs. If that TS compilation happens as part of the "new open" webpack build, then that happens automatically.

But, even without the followup work, I think the "clean separation" is worthwhile: We can develop new features without worrying about adverse affects to existing pages on Open.
